boot/freeldr/freeldr/lib/fs/fs.c

```diff
@@ -444,7 +444,7 @@ VOID FsRegisterDevice(CHAR* Prefix, const DEVVTBL* FuncTable)
     TRACE("FsRegisterDevice() Prefix = %s\n", Prefix);
 
     Length = strlen(Prefix) + 1;
-    pNewEntry = FrLdrTempAlloc(sizeof(DEVICE) + Length, TAG_DEVICE);
+    pNewEntry = FrLdrTempAlloc(sizeof(DEVICE) + (ULONG)Length, TAG_DEVICE);
```

Comment:
Wouldn't it be better if FrLdrTempAlloc took a SIZE_T as buffer size parameter instead?

Reply:
Yeah I guess that would make sense.